atlasdb-config/src/main/java/com/palantir/atlasdb/factory/AtlasDbDialogueServiceProvider.java

```diff
@@ -0,0 +1,142 @@
+                "Dialogue service provider attempted to provide a non-Dialogue class."
+                        + " This is an AtlasDB bug.");
+        ConjureTimelockServiceBlocking longTimeoutService
+                = dialogueClientFactory.get(ConjureTimelockServiceBlocking.class, TIMELOCK_LONG_TIMEOUT);
+        ConjureTimelockServiceBlocking shortTimeoutService
+                = dialogueClientFactory.get(ConjureTimelockServiceBlocking.class, TIMELOCK_SHORT_TIMEOUT);
+
+        ShortAndLongTimeoutServices<ConjureTimelockService> shortAndLongTimeoutServices
+                = ImmutableShortAndLongTimeoutServices.<ConjureTimelockServiceBlocking>builder()
+                .longTimeout(longTimeoutService)
+                .shortTimeout(shortTimeoutService)
+                .build()
+                .map(proxy -> FastFailoverProxy.newProxyInstance(ConjureTimelockServiceBlocking.class, () -> proxy))
+                .map(DialogueAdaptingConjureTimelockService::new);
+
+        return new TimeoutSensitiveConjureTimelockService(shortAndLongTimeoutServices);
+    }
+
+    private static RemoteServiceConfiguration createRemoteServiceConfiguration(
+            UserAgent userAgent, ServerListConfig serverListConfig, boolean shouldUseExtendedTimeout) {
+        return ImmutableRemoteServiceConfiguration.builder()
+                .remotingParameters(getFailoverRemotingParameters(shouldUseExtendedTimeout, userAgent))
+                .serverList(serverListConfig)
+                .build();
+    }
+
+    private static AuxiliaryRemotingParameters getFailoverRemotingParameters(
+            boolean shouldSupportBlockingOperations, UserAgent userAgent) {
+        return AuxiliaryRemotingParameters.builder()
+                .shouldLimitPayload(true)
+                .shouldRetry(true)
+                .shouldUseExtendedTimeout(shouldSupportBlockingOperations)
+                .userAgent(userAgent)
+                .build();
+    }
+
+    private static boolean isDialogue(Class<?> serviceInterface) {
+        return getStaticOfMethod(serviceInterface).isPresent();
+    }
+
+    private static Optional<Method> getStaticOfMethod(Class<?> dialogueInterface) {
+        try {
+            return Optional.ofNullable(dialogueInterface.getMethod("of", Channel.class, ConjureRuntime.class));
+        } catch (NoSuchMethodException e) {
+            return Optional.empty();
+        }
+    }
+
+    private static DialogueClients.ReloadingFactory decorateForFailoverServices(
+            DialogueClients.ReloadingFactory baseFactory,
+            Refreshable<Map<String, RemoteServiceConfiguration>> serviceToRemoteConfiguration) {
+        return baseFactory.reloading(serviceToRemoteConfiguration.map(
+                DialogueClientOptions::toServicesConfigBlock))
+                .withNodeSelectionStrategy(NodeSelectionStrategy.PIN_UNTIL_ERROR_WITHOUT_RESHUFFLE)
+                .withClientQoS(ClientConfiguration.ClientQoS.DANGEROUS_DISABLE_SYMPATHETIC_CLIENT_QOS);
```

Comment:
Is this new? In our initial testing I think we used the concurrency limiters without running into problems. In dialogue we've tried to allow them to ramp up more more quickly than CJR. There's definitely a bit of risk as we continue to tune the rate limiters, but it could help us avoid knocking over lock with too many handshakes.

Reply:
This was used in CJR-land as well, so it isn't new - I just ported it over. We should try reinstating the limiters at some point if the Dialogue ones are more aggressive, though I'd like to get the migration through first.